tls/s2n_tls13_secrets.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "tls/s2n_crypto_constants.h"
+#include "tls/s2n_quic_support.h"
+#include "utils/s2n_blob.h"
+#include "utils/s2n_result.h"
+
+/* tls/s2n_tls13_secrets should eventually replace crypto/s2n_tls13_keys.h */
+#include "crypto/s2n_tls13_keys.h"
+
+typedef enum {
+    S2N_EARLY_SECRET = 0,
+    S2N_HANDSHAKE_SECRET,
+    S2N_MASTER_SECRET
+} s2n_extract_secret_type_t;
+
+struct s2n_tls13_secrets {
```

Comment:
One general thing I'm still musing over with this PR is how this change interacts with TLS1.2. Is the tls1.2 master secret now in a different place than the tls1.3 master secret?

Reply:
It doesn't change the interaction. The two sets of secrets were [already split](https://github.com/aws/s2n-tls/blame/3a1970ec9a7d63b36c46646ff90c8cc19aeebe80/tls/s2n_crypto.h#L60-L63). We weren't even intentionally storing the TLS1.3 master secret before, just the "derived" and "extracted" secrets.

The two secrets also really only share a name, so there's no reason they have to be in the same place.